### PR TITLE
Add implementation details on Find Command

### PR DESCRIPTION
**Pull request by tenebrius1**
- Added relevant implementation details on FindCommand
- Added relevant UML Sequence and Activity Diagrams on FindCommand

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -203,9 +203,44 @@ This section describes some noteworthy details on how certain features are imple
...
+2. `LogicManager` calls on `AddressBookParser`'s `parseCommand()` method which in turns creates a new `FindCommandParser`.
+3. The `FindCommandParser` calls its own `parse()` method which will return a new `FindCommand` if the input is valid.
+4. `FindCommand` will then update the student list in the `Model` class.
+5. Lastly, a new `CommandResult` with the success message is returned to the `LogicManager`.
+
+Given below is a sequence diagram to show how the `find` implementation works:
+
+![FindCommandSequenceDiagram](images/FindCommandSequenceDiagram.png)
+
+Given below is an activity diagram to show how the `find` command works:
```
Comment (zihaooo9) [inline docs/DeveloperGuide.md:228]: Can mention that you are specifying the find command for a **valid** find student input

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -203,9 +203,44 @@ This section describes some noteworthy details on how certain features are imple
...
+
+![FindCommandActivityDiagram](images/FindCommandActivityDiagram.png)
+
+#### Design Considerations:
+
+##### Aspect: How `find` executes
+
+- **Alternative 1 (current choice)**: User can `find` using multiple prefixes at once.
+  - Pros: It allows the User to find a specific type of student or tutor more easily .
```
Comment (knotstoks) [inline docs/DeveloperGuide.md:237]: Minor nit of the extra space before the fullstop

File: docs/diagrams/FindCommandActivityDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+@startuml
+start
+:User executes find command;
+:Parse command arguments;
+
+if () then ([Invalid input/s])
+    :Throw \nParseException;
```
Comment (knotstoks) [inline docs/diagrams/FindCommandActivityDiagram.puml:7]: Perhaps you could kindly end it here

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -203,9 +203,44 @@ This section describes some noteworthy details on how certain features are imple
...
 
+##### Sequence of action
+{:.no_toc}
+
+Given below is an example usage scenario and how the `find` command implementation behaves at each step:
+
+1. The user input (for e.g `"find s n/David"`) is handled by the `CommandBox` class in the Ui component, before being passed to `LogicManager` to be executed.
+2. `LogicManager` calls on `AddressBookParser`'s `parseCommand()` method which in turns creates a new `FindCommandParser`.
+3. The `FindCommandParser` calls its own `parse()` method which will return a new `FindCommand` if the input is valid.
+4. `FindCommand` will then update the student list in the `Model` class.
```
Comment (zihaooo9) [inline docs/DeveloperGuide.md:221]: Perhaps you could specify the method used in the `Model` class (eg. `Model#updateStudentList`, not really sure what the name is can check 🥲 )

**Pull request merged**